File: src/main/java/com/soapdogg/lispInterpreter/tokenizer/LineTokenizer.java
package com.soapdogg.lispInterpreter.tokenizer;

import com.soapdogg.lispInterpreter.datamodels.Token;
import com.soapdogg.lispInterpreter.constants.TokenValueConstants;
import lombok.AllArgsConstructor;

import java.util.LinkedList;
import java.util.Queue;

@AllArgsConstructor(staticName = "newInstance")
public class LineTokenizer {

    private final WordTokenizer wordTokenizer;

    public Queue<Token> tokenizeLine(
        String line
    ) {
        String trimmedLine = line.trim();
        String [] words = trimmedLine.split(TokenValueConstants.WHITE_SPACE_PATTERN);
        Queue<Token> tokens = new LinkedList<>();
        for (String word : words) {
            Queue<Token> wordTokens = wordTokenizer.tokenizeWord(word);
            tokens.addAll(wordTokens);
        }
        return tokens;
    }
}

File: src/main/java/com/soapdogg/lispInterpreter/tokenizer/LineTokenizer.kt
package com.soapdogg.lispInterpreter.tokenizer

import com.soapdogg.lispInterpreter.constants.TokenValueConstants
import com.soapdogg.lispInterpreter.datamodels.Token

class LineTokenizer (
    private val wordTokenizer: WordTokenizer
){

    fun tokenizeLine(
        line: String
    ): List<Token> {
        val trimmedLine = line.trim { it <= ' ' }
        val words = trimmedLine.split(TokenValueConstants.WHITE_SPACE_PATTERN.toRegex())
        return words.map {
            word -> wordTokenizer.tokenizeWord(word)
        }.flatMap { it.toList() }
    }
}